src/palace/manager/feed/acquisition.py

```diff
@@ -633,22 +627,23 @@ def single_entry(
             elif work.presentation_edition:
                 active_edition = work.presentation_edition
                 identifier = active_edition.primary_identifier
+            else:
+                active_edition = None
 
         # There's no reason to present a book that has no active license pool.
         if not identifier:
-            logging.warning("%r HAS NO IDENTIFIER", work)
-            return None
+            raise PalaceValueError("Work has no identifier")
```

Comment:
Do we not want the work info in this log message? Or are we getting that somehow?

Reply:
Good call. I updated the exception to add the `repr` to the message:

```
PalaceValueError(f"Work has no associated identifier: {work!r}")
```